test-harness/Cargo.toml

```diff
@@ -8,5 +8,8 @@ edition = "2021"
 [features]
 emu = []
 
+# Whether or not this is a harness for a runtime firmware image
+runtime = []
```

Comment:
Can you explain why this is necessary?

Reply:
For the runtime tests I'm using the assembly files from `cpu/src` for early boot. This is the real entrypoint for runtime firmware, so I thought it was ideal to just use that.

For the driver tests, I kept it using the test `start.S` (which now lives in the test-harness crate). This is much simpler, it just sets up the memory regions and runs main.

So this feature is used to tell test-harness whether it is using the runtime early boot code. In that case, the test-harness crate needs to:

1. Set up some extra handlers
2. Doesn't need to include a start.S (it will be provided by the cpu crate)

If the caller doesn't want to use the runtime early boot code, the test-harness crate will just provide the simple `start.S`.

Reply:
It would potentially be acceptable to just make everyone use the boot code from the start.S crate. I was trying to avoid changing the behavior of the driver tests by keeping the same boot code, but I'm not sure it matters much.

Reply:
Are you booting to your tests through the ROM/FMC?

Reply:
Ya

Reply:
For any tests that you would want to use to debug the RTL (like the drivers), it's best that we keep the startup code as simple as possible, so the tests run in a reasonable amount of time, and the waveforms (needed to debug the RTL) don't get too big.

I might try doing a pass at some point to unify all this code, but it won't be using the normal boot ROM, instead just jumping straight to ICCM with no validation or anything.

Reply:
Sounds good. Thanks.